### ISSUE TITLE
Contact data isn't updated

### ISSUE THREAD

**Issue by AlfredSK** [friendica/friendica#11324]
- [X ] I have searched open and closed issues for duplicates

### Bug Description

Looks like the contact data of (remote) contacts isn't updated. A user of my node reported that some remote Mastodon contact is still showing a very old profile picture and also an outdated user name. I follow the same remote user as well and I can confirm that. The remote user changed their user (display) name and their user avatar some month ago but Friendica is still showing the old name and image.

Even re-fetching the contact data (contact -> action menu -> re-fetch contact data) doesn't help. Nothing is updated.

Actual Result:

Contact data isn't automatically updated. Manual update also doesn't work.

Expected Result:

Profile pictures and user names should show updated data automatically within 24 hours. A manual re-fetch of the contact data should update and show the correct contact data instantly.

### Platform Info

Friendica Version:

2022.05-dev

Friendica Source: 

git

PHP version:

7.4

SQL version:

10.6 MariaDB

**Comment by AlfredSK** [comment]
Report and discussion: https://libranet.de/display/0b6b25a8-1562-2c75-a48e-8e7017168483

**Comment by annando** [comment]
Have you checked with other contacts?

**Comment by AlfredSK** [comment]
> Have you checked with other contacts?

No. Not yet. I need to know someone who changed their name and avatar recently. Or I need to test it myself.

**Comment by AlfredSK** [comment]
Ok. I have changed the username and the profile picture of my mastodon.social account. I will see what happens with the contact on my Friendica node.

**Comment by AlfredSK** [comment]
After 15 seconds I have the new contact data on Friendica. Profile image and user name were automatically updated. Weird.

**Comment by FiXato** [comment]
After following Christine from my Libranet account just now, her profile name and picture seem to have updated.

**Comment by AlfredSK** [comment]
Works for me too now. Hmm.

**Comment by annando** [comment]
Air humidity?

**Comment by AlfredSK** [comment]
Probably. 😊 I'll watch this for some days before closing the issue.

**Comment by foss-** [comment]
Seeing missing avatars in contact list for <s>many</s> all twitter users I follow. However when selecting `View Contact` the avatar is there. Could this be related?
Unfollowing and re-following brings back the avatar image in contact list.

![Screenshot 2022-05-05 at 00 15 47](https://user-images.githubusercontent.com/5708172/166834644-18e2421c-67e8-4c95-996a-95120b7f8415.png)
![Screenshot 2022-05-05 at 00 16 10](https://user-images.githubusercontent.com/5708172/166834641-d9d90553-b979-42ec-86db-7b7306edb1f7.png)

**Comment by annando** [comment]
@foss- which server?

**Comment by foss-** [comment]
piratica

**Comment by annando** [comment]
Yeah, that's a downside of the configuration to avoid storing avatar data. I guess in the next release I will add some improvement there.

**Comment by foss-** [comment]
Not limited to avatar. As can be seen in the following example the username is also not updated when it is changed:
![Screenshot 2022-05-06 at 00 13 19](https://user-images.githubusercontent.com/5708172/167034007-d50fc3e1-5c0d-4455-ac01-a136eae9e9d5.png)
![Screenshot 2022-05-06 at 00 13 37](https://user-images.githubusercontent.com/5708172/167034033-cf324fe3-28b9-4d97-8d83-6ea7452ab562.png)

**Comment by MrPetovan** [comment]
This may be a connector-specific issue where contacts aren't updated past the first fetch.

**Comment by annando** [comment]
Yes. This is a totally different thing.

**Comment by foss-** [comment]
Was unsure initially as the problem seemed related. It has its own entry now: https://github.com/friendica/friendica/issues/11475
